File: frontend/src/store/api/result/dto/product/variant/GramophonesVariantOutputDTO.ts
import { ProductVariantOutputDTO } from "./ProductVariantOutputDTO";

export interface GramophonesVariantOutputDTO extends ProductVariantOutputDTO {
    color: string,
    gramophonesId: number
};

export function isGramophonesVariantOutputDTO(variant: ProductVariantOutputDTO): variant is GramophonesVariantOutputDTO {
    return "gramophonesId" in variant;
}

export function isGramophonesVariantOutputDTOArray(variants: ProductVariantOutputDTO[]): variants is GramophonesVariantOutputDTO[] {
    return variants.every(isGramophonesVariantOutputDTO);
}

File: backend/product-client/src/main/java/org/andante/gramophones/dto/GramophonesVariantOutputDTO.java
package org.andante.gramophones.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.andante.product.dto.ProductVariantOutputDTO;

@SuperBuilder
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class GramophonesVariantOutputDTO extends ProductVariantOutputDTO {

    private String color;
    private Long gramophonesId;
}
